tests/ui.rs

```diff
@@ -130,9 +149,11 @@ fn run_tests(
         }
     }
     config.program.args.push("-Zui-testing".into());
-    config.program.args.push("--target".into());
-    config.program.args.push(target.into());
 
+    if config.host != config.target {
```

Comment:
Why can't we always set `--target`?

Reply:
`*-dep` tests start passing double `--target` args to rustc which fails because of that. Better solution is update to 0.22.3 (which currently blocked by other error)

Reply:
But why is that only a problem with `host == target`...? Very strange.

Anyway, please add a comment in the code.

Reply:
It seems to be a problem with `host != target` as well, judging from the CI failures
```
error: Option 'target' given more than once
```

I think ui_test 0.22 may be too buggy for us, unfortunately. We'll probably have to wait for 0.24 as 0.23 suffers from https://github.com/oli-obk/ui_test/issues/238.